Cannot build when using as a library

Not exactly sure what is causing this issue but when adding this as a dependency (using `master` not `0.3.0`) I get this error:

```
   Compiling dlc-manager v0.3.0 (https://github.com/p2pderivatives/rust-dlc#0a4e29fd)
error[E0599]: no method named `write` found for struct `HashMap` in the current scope
  --> /home/ben/.cargo/git/checkouts/rust-dlc-b5770de66c51b951/0a4e29f/dlc-manager/src/chain_monitor.rs:26:1
   |
26 | impl_dlc_writeable!(ChainMonitor, { (watched_tx, writeable), (last_height, writeable), (last_block_hashes, { cb_writeable, write_vec, read_vec}) ...
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `HashMap<bitcoin::Txid, chain_monitor::ChannelInfo>`
   |
   = note: this error originates in the macro `field_write` which comes from the expansion of the macro `impl_dlc_writeable` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `HashMap<bitcoin::Txid, chain_monitor::ChannelInfo>: lightning::util::ser::Readable` is not satisfied
  --> /home/ben/.cargo/git/checkouts/rust-dlc-b5770de66c51b951/0a4e29f/dlc-manager/src/chain_monitor.rs:26:1
   |
26 | impl_dlc_writeable!(ChainMonitor, { (watched_tx, writeable), (last_height, writeable), (last_block_hashes, { cb_writeable, write_vec, read_vec}) ...
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `lightning::util::ser::Readable` is not implemented for `HashMap<bitcoin::Txid, chain_monitor::ChannelInfo>`
   |
   = help: the following other types implement trait `lightning::util::ser::Readable`:
             ()
             (A, B)
             (A, B, C)
             AcceptDlc
             AcceptedChannel
             AcceptedContract
             AdaptorInfo
             AnnouncementSignatures
           and 205 others
   = note: this error originates in the macro `field_read` which comes from the expansion of the macro `impl_dlc_writeable` (in Nightly builds, run with -Z macro-backtrace for more info)
```

The project I am trying to pull it into has `lightning 0.0.12` as a dependency as well, possibly the error could be coming from there but doesn't seem like that would cause the issue

--- a/dlc-manager/src/chain_monitor.rs
+++ b/dlc-manager/src/chain_monitor.rs
@@ -4,7 +4,8 @@ use std::collections::HashMap;
 
 use bitcoin::{Block, BlockHash, Transaction, Txid};
 use dlc_messages::ser_impls::{
-    read_ecdsa_adaptor_signature, read_vec, write_ecdsa_adaptor_signature, write_vec,
+    read_ecdsa_adaptor_signature, read_hash_map, read_vec, write_ecdsa_adaptor_signature,
+    write_hash_map, write_vec,
 };
 use lightning::ln::msgs::DecodeError;
 use lightning::util::ser::{Readable, Writeable, Writer};
@@ -23,7 +24,7 @@ pub struct ChainMonitor {
     pub(crate) last_block_hashes: Vec<BlockHash>,
 }
 
-impl_dlc_writeable!(ChainMonitor, { (watched_tx, writeable), (last_height, writeable), (last_block_hashes, { cb_writeable, write_vec, read_vec}) });
+impl_dlc_writeable!(ChainMonitor, { (watched_tx, { cb_writeable, write_hash_map, read_hash_map}), (last_height, writeable), (last_block_hashes, { cb_writeable, write_vec, read_vec}) });
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub(crate) struct ChannelInfo {

--- a/dlc-messages/src/ser_impls.rs
+++ b/dlc-messages/src/ser_impls.rs
@@ -7,6 +7,8 @@ use lightning::ln::msgs::DecodeError;
 use lightning::ln::wire::Type;
 use lightning::util::ser::{Readable, Writeable, Writer};
 use secp256k1_zkp::{ffi::ECDSA_ADAPTOR_SIGNATURE_LENGTH, EcdsaAdaptorSignature};
+use std::collections::HashMap;
+use std::hash::Hash;
 use std::io::Read;
 
 const MAX_VEC_SIZE: u64 = 1000000;
@@ -557,6 +559,42 @@ pub fn read_as_tlv<T: Type + Readable, R: ::std::io::Read>(
     // This retrieves the length, will be removed once oracle specs are updated.
     let _: BigSize = Readable::read(reader)?;
     Readable::read(reader)
+}
+
+/// Writes a [`HashMap`].
+pub fn write_hash_map<W: Writer, T, V>(
+    input: &HashMap<T, V>,
+    writer: &mut W,
+) -> Result<(), ::std::io::Error>
+where
+    T: Writeable,
+    V: Writeable,
+{
+    (input.len() as u64).write(writer)?;
+
+    for (key, value) in input.iter() {
+        key.write(writer)?;
+        value.write(writer)?;
+    }
+
+    Ok(())
+}
+
+/// Reads a [`HashMap`].
+pub fn read_hash_map<R: ::std::io::Read, T, V>(reader: &mut R) -> Result<HashMap<T, V>, DecodeError>
+where
+    T: Readable + Hash + Eq,
+    V: Readable,
+{
+    let len: u64 = Readable::read(reader)?;
+    let mut map = HashMap::new();
+    for _ in 0..len {
+        let key: T = Readable::read(reader)?;
+        let value: V = Readable::read(reader)?;
+        map.insert(key, value);
+    }
+
+    Ok(map)
 }
 
 impl_dlc_writeable_external!(Payout, payout, { (offer, writeable), (accept, writeable) });

